ddtrace/contrib/aiohttp/patch.py

```diff
@@ -1,40 +1,375 @@
+    def __init__(self, obj, pin, span, trace_headers, trace_context):
+        super().__init__(obj)
+        pin.onto(self)
+        self._self_span = span
+        self._self_trace_headers = trace_headers
+        self._self_trace_context = trace_context
+        self._self_have_context = False
 
-def patch():
+    @asyncio.coroutine
+    def _handle_response(self, coro):
+        try:
+            resp = yield from coro  # noqa: E999
+
+            if self._self_trace_headers:
+                tags = {hdr: resp.headers[hdr]
+                        for hdr in self._self_trace_headers
+                        if hdr in resp.headers}
+                self._self_span.set_tags(tags)
+
+            self._self_span.set_tag(ext_http.STATUS_CODE, resp.status)
+            self._self_span.error = int(500 <= resp.status)
+            return resp
+        except BaseException:
+            self._self_span.set_traceback()
+            raise
+        finally:
+            if not self._self_have_context or not self._self_trace_context:
+                self._self_span.finish()
+
+    # this will get when called without a context
+    @asyncio.coroutine
+    def __iter__(self):
+        resp = yield from self._handle_response(self.__wrapped__.__iter__())  # noqa: E999
+        return resp
+
+    if PY_35:
+        def __await__(self):
+            resp = yield from self._handle_response(  # noqa: E999
+                self.__wrapped__.__await__())
+            return resp
+
+        @asyncio.coroutine
+        def __aenter__(self):
+            self._self_have_context = True
+            resp = yield from self._handle_response(  # noqa: E999
+                self.__wrapped__.__aenter__())
+            return resp
+
+        @asyncio.coroutine
+        def __aexit__(self, exc_type, exc_val, exc_tb):
+            try:
+                resp = yield from self.__wrapped__.__aexit__(exc_type, exc_val,  # noqa: E999
+                                                             exc_tb)
+            finally:
+                if self._self_have_context and self._self_trace_context:
+                    self._self_span.finish()
+            return resp
+
+
+def _create_wrapped_request(method, enable_distributed, trace_headers,
+                            trace_context, func, instance, args, kwargs):
+    # Use any attached tracer if available, otherwise use the global tracer
+    pin = Pin.get_from(instance)
+
+    # bail on the tracing if not enabled.
+    if not pin.tracer.enabled:
+        return func(*args, **kwargs)
+
+    if method == 'REQUEST':
+        if 'method' in kwargs:
+            method = kwargs['method']
+        else:
+            method = args[0]
+
+        if 'url' in kwargs:
+            url = URL(kwargs['url'])
+        else:
+            url = URL(args[1])
+    else:
+        if 'url' in kwargs:
+            url = URL(kwargs['url'])
+        else:
+            url = URL(args[0])
+
+    if should_skip_request(pin, url):
```

Comment:
🤔 this isn't used and doesn't do anything 😆

Reply:
this will not trace the request similarly to how httplib's tracer skips tracing under certain circumstances

Reply:
oh, I'm not quite sure what I was thinking when I wrote this comment... I think I thought it was a function or something? lol disregard me here 😆